lib/wallet/src/Cardano/Wallet/Shelley/Network/Node.hs

```diff
@@ -354,7 +361,11 @@ withNodeNetworkLayerBase
     queryRewardQ <- connectDelegationRewardsClient
         (handlers ClientDelegationRewards)
 
-    rewardsObserver <- newRewardBalanceFetcher tr readNodeTip queryRewardQ
+    rewardsObserverFromKeyHash <- newRewardBalanceFetcher
+        tr readNodeTip queryRewardQ RewardAccountFromKeyHash
+    rewardsObserverFromScriptHash <- newRewardBalanceFetcher
+        tr readNodeTip queryRewardQ RewardAccountFromScriptHash
```

Comment:
Creating two different fetchers with two different `RewardAccountSource` is unnecessary. Batching all script and key credentials into the same local state query should be more efficient. The reason you can't currently do this is the `RewardAccount` type containing too little information.

What about:
1. Either importing or re-exporting `Cardano.StakeCredential` / `Ledger.StakeCredential crypto` or defining something similar to it. (Could also be `data StakeCredential = ...FromKeyHash RewardAccount | ...FromScriptHash  RewardAccount`)
2. Changing `getCachedRewardAccountBalance` to take `StakeCredential` instead of `Either RewardAccount RewardAccount`
3. Removing `RewardAccountSource` and the duplicate call to `newRewardBalanceFetcher`

?